e2e/bazel/bazel.go

```diff
@@ -0,0 +1,180 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	blaze_query "github.com/bazelbuild/bazel-watcher/third_party/bazel/master/src/main/protobuf"
+	"github.com/golang/protobuf/proto"
+)
+
+func fuzzyEqual(pattern []interface{}, compare []string) bool {
+	if len(pattern) != len(compare) {
+		return false
+	}
+
+	for k, _ := range pattern {
+		switch v := pattern[k].(type) {
+		case *regexp.Regexp:
+			if !v.MatchString(compare[k]) {
+				return false
+			}
+			break
+		case string:
+			if v != compare[k] {
+				return false
+			}
+			break
+		default:
+			fmt.Errorf("I have no idea what a %T is... please fix.", v)
+			panic(v)
+		}
+	}
+
+	return true
+}
+
+func runWithScript(args []string) {
+	// Since all ru ncommands start with "run" as arg0 and it breaks flag
```

Comment:
// All run commands start with "run" as the first argument while all other arguments are flags.